File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int n = nums.size() ;
        int l = 0, h = n - 1;
        int mini = INT_MAX;
        while(l <= h){
            int m = (l + h) >> 1;
            if(l == h) mini = min(mini, nums[l]);
            if(nums[l] <= nums[m]) {
                mini = min(mini, nums[l]);
                l = m + 1;
            } else {
                mini = min(mini, nums[m]);
                h = m - 1;
            }
        }
        return mini;
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, A: List[int]) -> int:
        l, h = 0 , len(A) - 1
        mini = float("inf")
        while l <= h:
            m = (l + h) >> 1
            if A[l] <= A[m]:
                mini = min(mini, A[l])
                l = m + 1
            else:
                mini = min(mini, A[m])
                h = m - 1
                
        return mini
                
        